Drupal 10 support

stephen-cox opened the pull request:
There needs to be a major release of LocalGov Geo fro Drupal support (https://github.com/localgovdrupal/localgov_geo/issues/65#issuecomment-1447093730), which as a dependency of Events, requires a major version bump for Events.

This will require a 3.x release.

Fixes #91

stephen-cox committed:
Drupal 10 support

stephen-cox committed:
Alias modules for Drupal 10 support

stephen-cox committed:
Patch date_recur_modular for Drupal 10 support

stephen-cox committed:
Use dev version of date_recur_modular

stephen-cox committed:
Use entity_manager service rather than generator when configuring simple_sitemap

stephen-cox committed:
Update tests to run on Drupal 9 or 10

ekes commented on composer.json:
```diff
@@ -6,12 +6,12 @@
-        "drupal/date_recur": "^3.0",
-        "drupal/date_recur_modular": "^3.0",
+        "drupal/date_recur": "^3.2",
+        "drupal/date_recur_modular": "3.x-dev#6ea45215",
```
Can already go 3.1.0-rc1 https://www.drupal.org/project/date_recur_modular/releases/3.1.0-rc1

stephen-cox replied:
Updated the reference - thanks

stephen-cox committed:
Use date_recur_modular 3.1@RC

stephen-cox committed:
Run GitHub Actions with Drupal 10

stephen-cox committed:
Show summary on event edit form #97 (#98)

ekes committed:
Fix #101 only install required optional dependencies. (#102)

Added dependencies from localgov_directories_page and
localgov_directories_venue.

stephen-cox committed:
Merge branch '2.x' of https://github.com/localgovdrupal/localgov_events into feature/drupal-10

ekes commented on localgov_events.info.yml:
```diff
@@ -1,6 +1,6 @@
 name: LocalGov Events
 description: Provides events for LocalGov Drupal.
-core_version_requirement: ^8.9 || ^9
+core_version_requirement: ^9 || ^10
```
This removes 9 from the matrix, probably because of the dependency on Geo, so probably shouldn't be claiming 9 compat here? Putting as a D10 only release in the maxtrix.

stephen-cox committed:
Update version for latest D10 support

finnlewis committed:
Fix coding standards.

finnlewis commented:
```
There was 1 error:

1) Drupal\Tests\localgov_events\Functional\EventPageTest::testEventFields
Exception: Warning: Undefined array key "localgov_directories_venue"
Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem::Drupal\Core\Field\Plugin\Field\FieldType\{closure}()() (Line: 144)
```

I can't replicate this manually locally. I enable localgov_events, then enable localgov_directories_venue and I can see the field. Maybe I'm barking up the wrong tree?

finnlewis commented:
Hmmm.. odd. When I comment out line 79, the test passes. 

https://github.com/localgovdrupal/localgov_events/blob/feature/drupal-10/tests/src/Functional/EventPageTest.php#L79

```
    // Check optional provider field.
    \Drupal::service('module_installer')->install(['localgov_directories_page']);
    // $this->drupalGet('/admin/structure/types/manage/localgov_event/fields');
    // $this->assertSession()->pageTextContains('localgov_event_provider');
    // $this->assertSession()->pageTextNotContains('localgov_event_venue');

    // Check optional venue field.
    \Drupal::service('module_installer')->install(['localgov_directories_venue']);
    $this->drupalGet('/admin/structure/types/manage/localgov_event/fields');
    $this->assertSession()->pageTextContains('localgov_event_venue');
```
Uncommenting line 79 makes the test fail again: 

```
    // Check optional provider field.
    \Drupal::service('module_installer')->install(['localgov_directories_page']);
    $this->drupalGet('/admin/structure/types/manage/localgov_event/fields');
    // $this->assertSession()->pageTextContains('localgov_event_provider');
    // $this->assertSession()->pageTextNotContains('localgov_event_venue');

    // Check optional venue field.
    \Drupal::service('module_installer')->install(['localgov_directories_venue']);
    $this->drupalGet('/admin/structure/types/manage/localgov_event/fields');
    $this->assertSession()->pageTextContains('localgov_event_venue');
```

Why is hitting the /admin/structure/types/manage/localgov_event/fields page triggering the error?
```Exception: Warning: Undefined array key "localgov_directories_venue"```

finnlewis commented:
Would be good to fix this test so we can release an alpha for Drupal 10 support.

stephen-cox committed:
Fixing tests

finnlewis reviewed (approved):
Fantastic, thanks!
